xarray/core/formatting.py

```diff
@@ -493,9 +507,14 @@ def array_repr(arr):
     else:
         name_str = ""
 
+    if _get_boolean_with_default("display_expand_data", default=True):
+        data_repr = short_data_repr(arr)
+    else:
+        data_repr = inline_variable_array_repr(arr, OPTIONS["display_width"])
```

Comment:
this raises for
```python
ds = xr.tutorial.open_dataset("rasm")
with xr.set_options(display_expand_data=False):
    display(ds.Tair)
```
i.e. with the data being a `MemoryCachedArray`. Should this be something like
```suggestion
        data_repr = inline_variable_array_repr(arr.variable, OPTIONS["display_width"])
```
or should we let `display_expand_data` have no effect for `MemoryCachedArray` (the expanded `repr` is already really small)? The difference would be: `[2029500 values with dtype=float64]` for `display_expand_data=True` vs `...` for `display_expand_data=False`

Reply:
Thanks a lot for spotting @keewis .

No strong view on which of those we go ahead with — @tomwhite are you familiar with this? Otherwise @keewis do you want to make a call?

@tomwhite how would you feel about following up your first PR with a second? :)

Reply:
I think it's more helpful to always have the expanded `repr`, but I'm not sure how complicated that special-case would be.

Reply:
Thanks for finding this @keewis. I can put together a PR - I'm just trying to find a simple test case that exposes the problem (the tests don't currently use the tutorial as far as I can tell).

Reply:
this should work with any opened file, so the test case should use either `open_dataset` or `open_dataarray` (creating the `MemoryCachedArray` by hand would also work but might be a bit more difficult). Something like
```python
@requires_netCDF4
def test_repr_memory_cached_collapsed(temp_path):
    arr = xr.DataArray([0, 1], dims="x")
    arr.to_netcdf(temp_path, engine="netcdf4")

    with xr.open_dataarray(temp_path) as arr, xr.set_options(display_expand_data=True):
        # check the result of repr
        ...
```
might work? (`temp_path` should be a builtin pytest fixture to create a temporary file and return the path, but I'm not sure I got the name right)

Reply:
I've opened #5235 with a possible fix.